### ISSUE TITLE
单独使用文本识别模型TextRecognition类的推理结果和使用命令行 predict_rec.py推理结果不一致

### ISSUE THREAD

**Issue by WangJ-oss** [PaddlePaddle/PaddleOCR#17276]
### 🔎 Search before asking

- [x] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.
- [x] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.
- [x] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### 🐛 Bug (问题描述)

我是仅微调PP-OCRv5_mobile_rec模型，在自己的训练集训练完成后使用 `python tools/eval.py -c configs/rec/PP-OCRv5/PP-OCRv5_mobile_rec.yml  -o Global.pretrained_model="./output/PP-OCRv5_mobile_rec/latest"` 进行评估，得出的acc在0.997，但当我使用`python tools/export_model.py -c configs/rec/PP-OCRv5/PP-OCRv5_mobile_rec.yml -o Global.pretrained_model="./output/PP-OCRv5_mobile_rec/best_accuracy"           Global.save_inference_dir="./inference/rec_ppocrv5/"` 转化为推理模型，再在部署时使用 `def load_model(modle_dir, model_name):
    model = TextRecognition(
        model_dir=modle_dir, 
        model_name=model_name,  
        input_shape=(3,48,320),
    )
    return model
`  推理测试出来的acc相差一个点左右。 期间模型文件，测试集均使用一致。另外，我训练的时候使用的配置文件中的image_shape一样是(3,48,320)。以下是我的配置文件： `Global:
  model_name: PP-OCRv5_mobile_rec # To use static model for inference.
  debug: false
  use_gpu: true
  epoch_num: 500
  log_smooth_window: 10
  print_batch_step: 100
  save_model_dir: ./output/v3_mix_2_11_29_PP-OCRv5_mobile_rec
  save_epoch_step: 10
  eval_batch_step: 100
  cal_metric_during_train: true
  pretrained_model: ./pretrain_model/PP-OCRv5_mobile_rec_pretrained.pdparams
  checkpoints: 
  save_inference_dir:
  use_visualdl: false
  infer_img: doc/imgs_words/ch/word_1.jpg
  character_dict_path: ./ppocr/utils/dict/ppocrv5_dict.txt
  max_text_length: &max_text_length 10
  infer_mode: false
  use_space_char: true
  distributed: true
  save_res_path: ./output/rec/predicts_ppocrv5.txt
  d2s_train_image_shape: [3, 48, 320]


Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    name: Cosine
    learning_rate: 0.0005
    warmup_epoch: 10
  regularizer:
    name: L2
    factor: 3.0e-05


Architecture:
  model_type: rec
  algorithm: SVTR_LCNet
  Transform:
  Backbone:
    name: PPLCNetV3
    scale: 0.95
  Head:
    name: MultiHead
    head_list:
      - CTCHead:
          Neck:
            name: svtr
            dims: 120
            depth: 2
            hidden_dims: 120
            kernel_size: [1, 3]
            use_guide: True
          Head:
            fc_decay: 0.00001
      - NRTRHead:
          nrtr_dim: 384
          max_text_length: *max_text_length

Loss:
  name: MultiLoss
  loss_config_list:
    - CTCLoss:
    - NRTRLoss:

PostProcess:  
  name: CTCLabelDecode

Metric:
  name: RecMetric
  main_indicator: acc

Train:
  dataset:
    name: MultiScaleDataSet
    ds_width: false
    data_dir: ./train_data/mix_2/train
    ext_op_transform_idx: 1
    label_file_list:
    - ./train_data/mix_2/train/train.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - RecConAug:
        prob: 0.5
        ext_data_num: 2
        image_shape: [48, 320, 3]
        max_text_length: *max_text_length
    - RecAug:
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  sampler:
    name: MultiScaleSampler
    scales: [[320, 32], [320, 48], [320, 64]]
    first_bs: &bs 128
    fix_bs: false
    divided_factor: [8, 16] # w, h
    is_training: True
  loader:                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  
    shuffle: true
    batch_size_per_card: *bs
    drop_last: true
    num_workers: 16
Eval:
  dataset:
    name: SimpleDataSet
    data_dir: ./train_data/mix_2/test
    label_file_list:
    - ./train_data/mix_2/test/test.txt
    transforms:
    - DecodeImage:
        img_mode: BGR
        channel_first: false
    - MultiLabelEncode:
        gtc_encode: NRTRLabelEncode
    - RecResizeImg:
        image_shape: [3, 48, 320]
    - KeepKeys:
        keep_keys:
        - image
        - label_ctc
        - label_gtc
        - length
        - valid_ratio
  loader:
    shuffle: false
    drop_last: false
    batch_size_per_card: 32
    num_workers: 16
`  

### 🏃‍♂️ Environment (运行环境)

训练环境：ubuntu paddlepaddle-gpu 3.2.0 显卡v100四张
测试环境  Windows paddleocr 3.3.2 cpu


### 🌰 Minimal Reproducible Example (最小可复现问题的Demo)

# 本地推理
from paddleocr import TextRecognition
def load_model(modle_dir, model_name):
    model = TextRecognition(
        model_dir=modle_dir, 
        model_name=model_name,  
        input_shape=(3,48,320),
    )
    return model

#命令行推理
python tools/infer/predict_rec.py --image_dir="train_data/mix_2/test/images/8110_ls.jpg" --rec_model_dir="./inference/new4/"

**Comment by hjiehudbandbegshth** [comment]
Hi! From what I’ve seen, when switching from the CLI script (predict_rec.py) to the Python API (TextRecognition), the accuracy drop is usually caused by mismatched preprocessing and CTC-decode defaults.

In the CLI, predict_rec.py uses:
```
NormalizeImage(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], scale=1/255.)

CTCLabelDecode(use_space_char=True)
```
These are automatically inherited from your training config, so the CLI and eval pipeline stay consistent.

But in the Python API, TextRecognition actually defaults to:
```
NormalizeImage(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])

use_space_char=False
```
Since these don’t match your training/eval settings, both the input distribution and the decoding behavior change, so losing around 1% accuracy is pretty normal.

- A fix you can try

You can manually set TextRecognition to use the same preprocessing and decode configs as the CLI, e.g.:
```
from paddleocr import TextRecognition
model = TextRecognition(
    model_dir="./inference/rec_ppocrv5/",
    model_name="inference.pdmodel",       # optional, will auto-detect
    input_shape=(3,48,320),
    pre_process_params={
        "mean": [0.485, 0.456, 0.406],
        "std":  [0.229, 0.224, 0.225],
        "scale": 1/255.0
    },
    use_space_char=True,                  # keep spaces in CTC decode
    batch_size=32                         # match eval batch size
)
# Then run inference the same way as in predict_rec.py
```

Hope this helps! I'm looking forward to your reply!

**Comment by WangJ-oss** [comment]
> batch_size=32                         # match eval batch size

首先，很感谢您可以回复我的问题！在您的回复中，提到图片预处理不一致导致结果不一致，这点我认同，当我尝试用您所给的TextRecognition手动设置的参数，运行结果确实     pre_process_params、 use_space_char、  batch_siz这三个参数是不存在的，我所用的paddleocr版本是3.3.2

**Comment by hjiehudbandbegshth** [comment]
> > batch_size=32                         # match eval batch size
> 
> 首先，很感谢您可以回复我的问题！在您的回复中，提到图片预处理不一致导致结果不一致，这点我认同，当我尝试用您所给的TextRecognition手动设置的参数，运行结果确实 pre_process_params、 use_space_char、 batch_siz这三个参数是不存在的，我所用的paddleocr版本是3.3.2

Sorry for the confusion earlier. In PaddleOCR 3.3.2, the TextRecognition API does not expose parameters to directly set mean/std or use_space_char, so it's not possible to match predict_rec.py results just by passing arguments.
However, the root cause is confirmed. The accuracy difference does come from the preprocessing mismatch between TextRecognition (default mean/std = 0.5) and the training/CLI pipeline. To make the results fully consistent, you may need to adjust the internal preprocessing settings inside TextRecognition.

**Comment by WangJ-oss** [comment]
> > > batch_size=32 # 匹配评估批次大小
> > 
> > 
> > 首先，很感谢您可以回复我的问题！在您的回复中，提到图片预处理不一致导致结果不一致，这点我认同，当我尝试用您所给的TextRecognition手动设置的参数，运行结果确实 pre_process_params、 use_space_char、 batch_siz这三个参数是不存在的，我所用的paddleocr版本是3.3.2
> 
> 抱歉之前让你困惑了。在 PaddleOCR 3.3.2 中，TextRecognition API 不会直接暴露参数给设置的平均值/标准值或use_space_char，因此仅通过传递参数无法匹配predict_rec.py结果。然而，根本原因已被确认。准确率差异确实来自TextRecognition（默认平均值/标准值=0.5）与训练/CLI流水线之间的预处理不匹配。为了让结果完全一致，你可能需要调整TextRecognition中的内部预处理设置。

您好，再次打扰到您，我查看了predict_rec.py 确实有你说的NormalizeImage(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225], scale=1/255.) 处理代码，其函数为 

`   
 def resize_norm_img_abinet(self, img, image_shape):
        imgC, imgH, imgW = image_shape

        resized_image = cv2.resize(img, (imgW, imgH), interpolation=cv2.INTER_LINEAR)
        resized_image = resized_image.astype("float32")
        resized_image = resized_image / 255.0

        mean = np.array([0.485, 0.456, 0.406])
        std = np.array([0.229, 0.224, 0.225])
        resized_image = (resized_image - mean[None, None, ...]) / std[None, None, ...]
        resized_image = resized_image.transpose((2, 0, 1))
        resized_image = resized_image.astype("float32")

        return resized_image
`   
但是这是当rec_algorithm == "ABINet" ， 但是我cli命令为 python3 tools/infer/predict_rec.py   --image_dir="./train_data/mix/test/images/33957_xycs.jpg"     --rec_model_dir="./inference/new4/"   --rec_image_shape="3,48,320"  其中没有明确指定 rec_algorithm  即使用默认的 SVTR_LCNet，该算法也是我训练时所用到的。 总结，还请麻烦您再看看这个问题

**Comment by WangJ-oss** [comment]
> > > batch_size=32                         # match eval batch size
> > 
> > 
> > 首先，很感谢您可以回复我的问题！在您的回复中，提到图片预处理不一致导致结果不一致，这点我认同，当我尝试用您所给的TextRecognition手动设置的参数，运行结果确实 pre_process_params、 use_space_char、 batch_siz这三个参数是不存在的，我所用的paddleocr版本是3.3.2
> 
> Sorry for the confusion earlier. In PaddleOCR 3.3.2, the TextRecognition API does not expose parameters to directly set mean/std or use_space_char, so it's not possible to match predict_rec.py results just by passing arguments. However, the root cause is confirmed. The accuracy difference does come from the preprocessing mismatch between TextRecognition (default mean/std = 0.5) and the training/CLI pipeline. To make the results fully consistent, you may need to adjust the internal preprocessing settings inside TextRecognition.

您好非常感谢你前面的指导， 我现在有一个新的问题，我已经成功让predict_rec.py的推理结果和 TextRecongition类推理结果一致，但是我发现精度还是不对，`python tools/export_model.py -c configs/rec/PP-OCRv5/PP-OCRv5_mobile_rec.yml -o Global.pretrained_model="./output/PP-OCRv5_mobile_rec/best_accuracy"           Global.save_inference_dir="./inference/rec_ppocrv5/"` 评估模型时的精度和我讲模型转化为推理模型后再用TextRecongition类推理的精度值不一致 请问是哪里出现问题了呢

**Comment by hjiehudbandbegshth** [comment]
> > > > batch_size=32                         # match eval batch size
> > > 
> > > 
> > > 首先，很感谢您可以回复我的问题！在您的回复中，提到图片预处理不一致导致结果不一致，这点我认同，当我尝试用您所给的TextRecognition手动设置的参数，运行结果确实 pre_process_params、 use_space_char、 batch_siz这三个参数是不存在的，我所用的paddleocr版本是3.3.2
> > 
> > 
> > Sorry for the confusion earlier. In PaddleOCR 3.3.2, the TextRecognition API does not expose parameters to directly set mean/std or use_space_char, so it's not possible to match predict_rec.py results just by passing arguments. However, the root cause is confirmed. The accuracy difference does come from the preprocessing mismatch between TextRecognition (default mean/std = 0.5) and the training/CLI pipeline. To make the results fully consistent, you may need to adjust the internal preprocessing settings inside TextRecognition.
> 
> 您好非常感谢你前面的指导， 我现在有一个新的问题，我已经成功让predict_rec.py的推理结果和 TextRecongition类推理结果一致，但是我发现精度还是不对，`python tools/export_model.py -c configs/rec/PP-OCRv5/PP-OCRv5_mobile_rec.yml -o Global.pretrained_model="./output/PP-OCRv5_mobile_rec/best_accuracy" Global.save_inference_dir="./inference/rec_ppocrv5/"` 评估模型时的精度和我讲模型转化为推理模型后再用TextRecongition类推理的精度值不一致 请问是哪里出现问题了呢

Hi, thanks again for your detailed follow-up!

First of all, to clarify:
The inconsistency you previously observed between TextRecognition and predict_rec.py was indeed caused by preprocessing differences. Since you’ve already confirmed they now produce consistent results, that issue is considered resolved.
The accuracy gap you’re seeing now is a different layer of the problem:



The main reason for  eval.py report higher accuracy than the exported inference model  is that the model structure used during training/evaluation is not the same as the one used after export.
During training and evaluation, the model runs with MultiHead (CTC + NRTR)
However, when exporting with export_model.py, the inference model keeps only the CTCHead
Because of this architecture change, it's quite normal for the exported model to show ~0.5%–1.5% lower accuracy compared to eval.py.

If you wish to make the results more consistent, you could try:
Training or evaluating with CTCHead only, so the evaluation structure matches the inference structure.
Or it might also be acceptable to keep the current model, considering that a ~1% accuracy difference is expected after export.
